src/Meshes/geoMesh.ts

```diff
@@ -0,0 +1,790 @@
+            this._isoVecsBAOA.push(temp);
+        }
+        
+    };
+
+    public _MapToFace(faceNb: number, geoData: PolyhedronData) {
+        const F = this._IDATA.face[faceNb];
+        const Oidx = F[2];
+        const Aidx = F[1];
+        const Bidx = F[0];
+    
+        const O = Vector3.FromArray(this._IDATA.vertex[Oidx]);
+        const A = Vector3.FromArray(this._IDATA.vertex[Aidx]);
+        const B = Vector3.FromArray(this._IDATA.vertex[Bidx]);
+    
+        const OA = A.subtract(O);
+        const OB = B.subtract(O);
+    
+        let x: Vector3 = OA.scale(this._coau).add(OB.scale(this._cobu));
+        let y: Vector3 = OA.scale(this._coav).add(OB.scale(this._cobv));
+        
+        const mapped = [];
+    
+        let idx: string; 
+        let tempVec: Vector3 = TmpVectors.Vector3[0]
+        for (var i = 0; i < this._cartesian.length; i++) {
+            tempVec  = x.scale(this._cartesian[i].x).add(y.scale(this._cartesian[i].y)).add(O);
+            mapped[i] = [tempVec.x, tempVec.y, tempVec.z];
+            idx = faceNb + "|" + this._vertices[i].x + "|" + this._vertices[i].y;
+            geoData.vertex[this._vecToIdx[idx]] = [tempVec.x, tempVec.y, tempVec.z];
+        }
+    };
+
+
+    //statics
+    /**Creates a primary triangle
+     * @param m 
+     * @param n 
+     * @hidden
+     */
+
+    public build(m: number, n: number) {
+        const vertices = new Array<IsoVector>();
+    
+        const O:IsoVector = IsoVector.Zero();
+        const A:IsoVector = new IsoVector(m, n);
+        const B:IsoVector = new IsoVector(-n, m + n);
+        vertices.push(O, A, B);
+    
+        //max internal isoceles triangle vertices
+        for (let y = n; y < m + 1 ; y++) {
+            for (let x = 0; x < m + 1 - y; x++ ) {
+                vertices.push(new IsoVector(x, y));
+            }
+        }
+    
+        //shared vertices along edges when needed
+        if (n > 0) {
+            const g = Scalar.HCF(m, n);
+            const m1 = m / g;
+            const n1 = n / g; 
+    
+            for (let i = 1; i < g; i++) {
+                vertices.push(new IsoVector(i * m1, i * n1)); //OA
+                vertices.push(new IsoVector(-i * n1, i * (m1 + n1)));  //OB
+                vertices.push(new IsoVector(m - i * (m1 + n1), n + i * m1)); // AB
+            }; 
+    
+            //lower rows vertices and their rotations
+            const ratio = m / n;
+            for (let y = 1; y < n; y++) {
+                for (let x = 0; x < y * ratio; x++) {
+                    vertices.push(new IsoVector(x, y));
+                    vertices.push(new IsoVector(x, y).rotate120(m , n));
+                    vertices.push(new IsoVector(x, y).rotateNeg120(m , n));
+                }
+            }
+        }
+        //order vertices by x and then y
+        vertices.sort((a, b) => {
+            return a.x - b.x
+        });
+    
+        vertices.sort((a, b) => {
+            return a.y - b.y
+        });
+    

```

Comment:
Just to be sure, you didn't want to do this instead of the two sorts:
```typescript
        vertices.sort((a, b) => {
            return a.x === b.x ? a.y - b.y : a.x - b.x;
        });
```
?

Reply:
The two sorts don't do the same thing than the single sort.

For eg, if you have (1,2), (0,3), (2,1):
* first method, sort the set by X then sort the set by Y:
  * sort by X: (0,3), (1,2), (2,1)
  * sort by Y: (2,1), (1,2), (0,3)
* second method, sort by X and for same X sort by Y:
  * (0,3), (1,2), (2,1)

It's not the same result.

Sorting the whole set by X then doing a second pass to sort by Y looked strange to me because I have never really saw this before and I don't understand what's the expected output (the X sort you do first will be in part wrecked by the Y sort depending on how the Y values are distributed - in my example above, the two methods lead to inverted sorting!), that's why I asked if what you wanted to do wasn't what I presented.

Reply:
OK understand. Think rows and columns. The two sorts sort the columns in order then the rows. Then for each row you read in order you will read the columns in order.